sklearn/datasets/_samples_generator.py

```diff
@@ -592,9 +600,29 @@ def make_regression(n_samples=100, n_features=100, *, n_informative=10,
 
     y = np.squeeze(y)
 
-    if coef:
+    if coef and not as_frame:
         return X, y, np.squeeze(ground_truth)
 
+    elif as_frame and not coef:
+        feature_names = [f'feat_{fn}' for fn in range(n_features)]
+        target_columns = [f'target_{tc}' for tc in range(n_targets)]
+        df, data, target = _convert_data_dataframe('make_classification',
+                                                   X,
+                                                   y,
+                                                   feature_names,
+                                                   target_columns)
+        return df, data, target
+
+    elif coef and as_frame:
+        feature_names = [f'feat_{fn}' for fn in range(n_features)]
+        target_columns = [f'target_{tc}' for tc in range(n_targets)]
+        df, data, target = _convert_data_dataframe('make_classification',
+                                                   X,
+                                                   y,
+                                                   feature_names,
+                                                   target_columns)
+        return df, data, target, np.squeeze(ground_truth)
```

Comment:
Could you please avoid such code duplication by using a nested `if coef / else` block under an `if as_frame / else` block instead?

```

Reply:
definetely. Thanks!